### PR TITLE
fix: windows functionality and tests

### PR DESCRIPTION
**Pull request by peakschris**
There are four issues when running on windows:

1. https://github.com/bazel-contrib/rules_bazel_integration_test/issues/330
2. https://github.com/bazel-contrib/rules_bazel_integration_test/issues/331
3. https://github.com/bazel-contrib/rules_bazel_integration_test/issues/332
4. https://github.com/bazel-contrib/rules_bazel_integration_test/issues/333

This PR fixes 3 and 4.

There are related PRs in bazel-skylib and bazel-starlib. There is no dependency -- the PRs can close in any order.
- https://github.com/cgrindel/bazel-starlib/pull/446 (fixes 1)
- https://github.com/bazelbuild/bazel-skylib/pull/527 (fixes 2)

### Test results:
Before:
--enable_runfiles: 0 pass
--noenable_runfiles: 0 pass

After this PR:
--enable_runfiles: 52 pass, 19 failures
--noenable_runfiles: 51 pass, 20 failures (17 are doc diff-tests due to bazel-starlib)

After this PR, together with wip PRs for 1 and 2:
--enable_runfiles: 71 pass, 0 failures
--noenable_runfiles: 51 pass, 20 failures (all due to bazel-starlib)

### CODE REVIEW

File: tests/tools_tests/setup_test_workspace.sh [hunk 1]
```diff
@@ -48,6 +48,9 @@ bazelrc_template="
 build --deleted_packages=
 query --deleted_packages=
 # EOF"
+# todo: should these lines be in the template above?
+# startup --windows_enable_symlinks
+# common --enable_runfiles
```
Comment (cgrindel) [inline tests/tools_tests/setup_test_workspace.sh:53]: I presume that we will determine the answer before we merge this PR.

Comment (peakschris) [reply tests/tools_tests/setup_test_workspace.sh:53]: Yes - can you help me understand that template? Is it used only for test cases internal to this repo or tests outside? Do you have a view as to whether these flags should be in here?

Comment (peakschris) [reply tests/tools_tests/setup_test_workspace.sh]: I've removed this

**Comment by cgrindel** [comment]
I approved the CI.

**Review by cgrindel — APPROVED** [review]

**Pull request merged**